### PR TITLE
feat: add GitHub component

### PR DESCRIPTION
**Pull request by YCK1130**
# Because

- We need a GitHub component to complete some development automation tasks

# This commit

## Related Issue
instill-ai/instill-core#1025

## Todo
- [X] TASK_GET_ALL_PULL_REQUESTS: function to get all prs given owner name and repository name.
- [X] TASK_GET_PULL_REQUEST: function to get a specific pr given owner name, repository name, and pr number. (including file changes)
- [X] TASK_GET_REVIEW_COMMENT: get review comment inside a pull request
- [X] TASK_CREATE_REVIEW_COMMENT: create review comment inside a pull request
- [X] TASK_GET_COMMIT: get commit messages and file changes
- [x] TASK_CREATE_ISSUE: post issue
- [x] TASK_GET_ALL_ISSUES: get all issues in a repo
- [x] TASK_GET_ISSUE: get an issue
- [x] TASK_CREATE_WEBHOOK: register webhook, https://docs.github.com/en/webhooks/webhook-events-and-payloads

### CODE REVIEW

File: operator/github/v0/README.mdx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_ALL_PULL_REQUESTS` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
+| State (required) | `state` | string | State of the PRs, including open, closed, all. Default: open |
+| Sort (required) | `sort` | string | Sort the PRs by created, updated, popularity, or long-running. Default: created |
+| Direction (required) | `direction` | string | Direction of the sort, including asc or desc. Default: asc |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:60]: We don't need to specify default values for required arguments as the user will have to set these themselves. Are these meant to be optional arguments?

File: operator/github/v0/README.mdx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+### Get All Pull Requests
+
+Get the list of all pull requests in a repository
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_ALL_PULL_REQUESTS` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:57]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+| Pull Requests (optional) | `pull_requests` | array[object] | An array of PRs |
+
+
+
+
+
+
+### Get Pull Request
+
+Get a pull request from a repository, given the PR number. Default is the latest PR
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:75]:
```suggestion
Get a pull request from a repository, given the PR number. This will default to the latest PR if no PR number is provided.
```

File: operator/github/v0/README.mdx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_PULL_REQUEST` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:82]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+Get a commit from a repository, given the commit SHA
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_COMMIT` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
+| Commit SHA (required) | `sha` | string | sha of the commit |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:106]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
| Commit SHA (required) | `sha` | string | SHA of the commit |
```

File: operator/github/v0/README.mdx [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+### Get Review Comments
+
+Get the review comments in a pull request
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_REVIEW_COMMENTS` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:128]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+| PR Number | `pr_number` | integer | Number of the PR. Default is the latest PR |
+| Sort | `sort` | string | Sort the comments by created, updated. Default: created |
+| Direction | `direction` | string | Direction of the sort, including asc or desc. Default: asc |
+| Since | `since` | string | Only comments updated at or after this time are returned. Default: 2021-01-01T00:00:00Z |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:132]:
```suggestion
| Sort | `sort` | string | Sort the comments by created, updated. Default is created |
| Direction | `direction` | string | Direction of the sort, including asc or desc. Default is asc |
| Since | `since` | string | Only comments updated at or after this time are returned. Default is 2021-01-01T00:00:00Z |
```

File: operator/github/v0/README.mdx [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+### Create Review Comment
+
+Create a review comment in pull request.
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_CREATE_REVIEW_COMMENT` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:154]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+### Get All Issues
+
+Get the list of all issues in a repository
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_ALL_ISSUES` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:178]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+| State | `state` | string | State of the issues, including open, closed, all. Default: open |
+| Sort | `sort` | string | Sort the issues by created, updated, popularity, or long-running. Default: created |
+| Direction | `direction` | string | Direction of the sort, including asc or desc. Default: asc |
+| Since | `since` | string | Only issues updated at or after this time are returned. Default: 2021-01-01T00:00:00Z |
+| No Pull Request | `no_pull_request` | boolean | Whether to not include pull requests in the issues. Default: false |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:183]:
```suggestion
| State | `state` | string | State of the issues, including open, closed, all. Default is open |
| Sort | `sort` | string | Sort the issues by created, updated, popularity, or long-running. Default is created |
| Direction | `direction` | string | Direction of the sort, including asc or desc. Default is asc |
| Since | `since` | string | Only issues updated at or after this time are returned. Default is 2021-01-01T00:00:00Z |
| No Pull Request | `no_pull_request` | boolean | Whether to not include pull requests in the issues. Default is false |
```

File: operator/github/v0/README.mdx [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+### Get Issue
+
+Get an issue.
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_GET_ISSUE` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:205]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+| Issue Number (required) | `issue_number` | integer | Number of the issue. |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:206]:
```suggestion
| Issue Number (required) | `issue_number` | integer | Number of the issue |
```

File: operator/github/v0/README.mdx [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+### Create Issue
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_CREATE_ISSUE` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
+| Issue title (required) | `title` | string | title of the issue |
+| Issue body (required) | `body` | string | body of the issue |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:228]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
| Issue title (required) | `title` | string | Title of the issue |
| Issue body (required) | `body` | string | Body of the issue |
```

File: operator/github/v0/README.mdx [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+
+
+### Create Webhook
+
+
+| Input | ID | Type | Description |
+| :--- | :--- | :--- | :--- |
+| Task ID (required) | `task` | string | `TASK_CREATE_WEBHOOK` |
+| Owner (required) | `owner` | string | owner of the repository |
+| Repository (required) | `repository` | string | repository name |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:248]:
```suggestion
| Owner (required) | `owner` | string | Owner of the repository |
| Repository (required) | `repository` | string | Repository name |
```

File: operator/github/v0/README.mdx [hunk 15]
```diff
@@ -0,0 +1,266 @@
...
+| Webhook URL (required) | `hook_url` | string | URL to send the payload to |
+| Events (required) | `events` | array[string] | Events to trigger the webhook. Please see https://docs.github.com/en/webhooks/webhook-events-and-payloads for more information |
+| Active | `active` | boolean | Whether the webhook is active |
+| Content Type | `content_type` | string | Content type of the webhook |
```
Comment (GeorgeWilliamStrong) [inline operator/github/v0/README.mdx:252]: Please add default values for optional arguments

**Review by GeorgeWilliamStrong — CHANGES_REQUESTED** [review]
Nice job with the documentation, just left a few minor comments for you to address.

File: application/github/v0/config/tasks.json [hunk 16]
```diff
@@ -0,0 +1,1607 @@
...
+            "value",
+            "reference"
+          ],
+          "instillUIOrder": 11,
+          "type": "string"
+        },
+        "direction": {
+          "default": "desc",
+          "title": "Direction",
+          "description": "Direction of the sort, including asc or desc. Default is asc",
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:780]: NIT
Typo?
`"description": "Direction of the sort, including asc or desc. Default is asc",`
Default is desc

File: application/github/v0/config/tasks.json [hunk 17]
```diff
@@ -0,0 +1,1607 @@
...
+        }
+      },
+      "required": [
+        "pull_requests"
+      ],
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_GET_PULL_REQUEST": {
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:827]: Same Question above
I am curious about the purpose of the design here.
If we remain multiple one with params `pr_number`, will it be cleaner in terms of UX / codebase?

File: application/github/v0/config/tasks.json [hunk 18]
```diff
@@ -0,0 +1,1607 @@
...
+            "value",
+            "reference"
+          ],
+          "instillUIOrder": 11,
+          "type": "string"
+        },
+        "direction": {
+          "default": "desc",
+          "title": "Direction",
+          "description": "Direction of the sort, including asc or desc. Default is asc",
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:930]: NIT
Typo?
`Default is asc",`

`Default is desc",`

File: application/github/v0/config/tasks.json [hunk 19]
```diff
@@ -0,0 +1,1607 @@
...
+            "integer"
+          ],
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "instillUIOrder": 3,
+          "type": "integer"
+        },
+        "comment": {
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:1021]: Memo
Not sure the design here.
Why do the users need to insert commit_id, path, ... so many params?

Will come back to check it after reading more code as well.

File: application/github/v0/config/tasks.json [hunk 20]
```diff
@@ -0,0 +1,1607 @@
...
+            "value",
+            "reference"
+          ],
+          "instillUIOrder": 11,
+          "type": "string"
+        },
+        "direction": {
+          "default": "desc",
+          "title": "Direction",
+          "description": "Direction of the sort, can be one of: asc, desc. Default is asc",
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:1194]: NIT
Typo?
`Default is asc"`

File: application/github/v0/config/tasks.json [hunk 21]
```diff
@@ -0,0 +1,1607 @@
...
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "instillUIOrder": 13,
+          "type": "string"
+        },
+        "no_pull_request": {
+          "title": "No Pull Request",
+          "description": "Whether to not include pull requests in the issues. Default is false",
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:1227]: Question
What if it is true?
I am not sure if the output in json deals with this case

Will come back to check it after reading more code again as well.

File: application/github/v0/config/tasks.json [hunk 22]
```diff
@@ -0,0 +1,1607 @@
...
+        }
+      },
+      "required": [
+        "issues"
+      ],
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_GET_ISSUE": {
```
Comment (chuang8511) [inline application/github/v0/config/tasks.json:1269]: Same question above.
If we remain multiple one with params `issue_number`, will it be cleaner in terms of UX / codebase?

File: application/github/v0/commits.go [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+	GetCommit(context.Context, string, string, string, *github.ListOptions) (*github.RepositoryCommit, *github.Response, error)
+	CreateHook(context.Context, string, string, *github.Hook) (*github.Hook, *github.Response, error)
+}
+
+type Commit struct {
+	SHA          	string         		`json:"sha"`
+	Message 		string          	`json:"message"`
+	Stats        	CommitStats         `json:"stats"`
+	Files 	  		[]CommitFile        `json:"files"`
+}
```
Comment (chuang8511) [inline application/github/v0/commits.go:21]: NIT
```
type Commit struct {
	SHA          	string         		`json:"sha"`
	Message 		string          	`json:"message"`
	Stats        	CommitStats         `json:"stats"`
	Files 	  		[]CommitFile        `json:"files"`
}
```
Please format it

File: application/github/v0/commits.go [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+			Changes: file.GetChanges(),
+		},
+	}
+}
+func (githubClient *Client) extractCommitInformation(originalCommit *github.RepositoryCommit) Commit {
+	stats := originalCommit.GetStats()
+	commitFiles := originalCommit.Files
+
+	if stats == nil || commitFiles == nil{
+		// fmt.Println("Getting stats for commit: ", originalCommit.GetSHA())
```
Comment (chuang8511) [inline application/github/v0/commits.go:48]: NIT
Please take it out.
```
// fmt.Println("Getting stats for commit: ", originalCommit.GetSHA())
```

File: application/github/v0/pull_request.go [hunk 25]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+func (githubClient *Client) getAllPullRequestsTask(props *structpb.Struct) (*structpb.Struct, error) {
+	err := githubClient.setTargetRepo(props)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &github.PullRequestListOptions{
+		State: props.GetFields()["state"].GetStringValue(),
```
Comment (chuang8511) [inline application/github/v0/pull_request.go:83]: Important
I think making dependency clean is important.

Could you take a look https://github.com/instill-ai/component/pull/187#discussion_r1662946437?

TLDR
I want to make protobuf struct to component struct.
In our other services, we depend on component struct rather than protobuf struct.

File: application/github/v0/pull_request.go [hunk 26]
```diff
@@ -0,0 +1,174 @@
...
+		}
+		resp.Commits = make([]Commit, len(commits))
+		for idx, commit := range commits {
+			resp.Commits[idx] = githubClient.extractCommitInformation(commit)
+		}
+	}
+	return resp, nil
+}
+
+// for testing
```
Comment (chuang8511) [inline application/github/v0/pull_request.go:64]: NIT
If it is only for test, we probably want it to be in the test file.

**Review by chuang8511 — COMMENTED** [review]
I finished my first review.
It is a big PR.
We may miss some details.

So, let's make it solid by building several pipelines to confirm if it works in each use case!

File: application/github/v0/pull_request.go (hunk 25 above)
Comment (YCK1130) [reply application/github/v0/pull_request.go]: Got it! I have made all tasks to component struct

File: application/github/v0/pull_request.go (hunk 26 above)
Comment (YCK1130) [reply application/github/v0/pull_request.go]: Due to [this change](https://github.com/instill-ai/component/pull/177#discussion_r1664228934), it's not only for test now.

File: application/github/v0/config/definition.json [hunk 27]
```diff
@@ -0,0 +1,27 @@
+{
+  "availableTasks": [
+    "TASK_GET_ALL_PULL_REQUESTS",
+    "TASK_GET_PULL_REQUEST",
+    "TASK_GET_COMMIT",
+    "TASK_GET_REVIEW_COMMENTS",
+    "TASK_CREATE_REVIEW_COMMENT",
+    "TASK_GET_ALL_ISSUES",
+    "TASK_GET_ISSUE",
```
[the remark this answers is not in the document]
Comment (YCK1130) [reply application/github/v0/config/definition.json:9]: I make tasks to get a specific pr/issue to simplify the response. If the client can only use multiple ones, they may need to use another component to filter out the response they want and make the pipeline more complex. 

Moreover, some detailed information may not be provided in the multiple ones' responses. The purpose of build "get all" here is to provide a tool for users to know what pr/issue the repo has if they don't know the pr/issue number.

[the remark this answers is not in the document]
Comment (YCK1130) [reply application/github/v0/config/definition.json:9]: Maybe we can combine the two multiple tasks into a "TASK_GET_REPO_INFO" and return all the info users might want to know?

File: application/github/v0/config/tasks.json (hunk 19 above)
Comment (YCK1130) [reply application/github/v0/config/tasks.json:1021]: Because we need to specify what commit, file, and the range of lines to create a comment.

File: application/github/v0/config/tasks.json (hunk 21 above)
Comment (YCK1130) [reply application/github/v0/config/tasks.json:1227]: If it is true, it will filter out all items that are actually a pr. Since issue and pr use the same indexing system in github, the API returns all relevant objects (issues and pr). I leave this param for user to simplify the response.

**Comment by YCK1130** [comment]
Concerns about this component

1. The rate limit of GitHub API calls for unauthenticated requests is 60 requests per hour, which is quite small. I left the token field optional because some tasks are available without authentication.
* Personal access token rate limit: 5000/hr
* GitHub Enterprise Cloud organization: 15000/hr
2. The "create review comment" task may need more checks. The [docs](https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#create-a-review-comment-for-a-pull-request) of GitHub are quite unclear and there seems to be some mismatch between the document and the actual API call.
3. I am using a Mock server to simulate the API calls just like in other components. I don't know whether it will be difficult to maintain in the future.

File: application/github/v0/config/definition.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (chuang8511) [reply application/github/v0/config/definition.json:9]: > Maybe we can combine the two multiple tasks into a "TASK_GET_REPO_INFO" and return all the info users might want to know?

I think your original way will be fine!
Aligning the vendors could bring as benefit in terms of workflow use case in my opinion.

It may be too various to know what users want to know in the cases of workflow because every company has different flows even when there are best practices.

File: application/github/v0/main.go [hunk 28]
```diff
@@ -0,0 +1,179 @@
...
+			err,
+			"Failed to unmarshal input",
+		)
+	}
+	inputMap := taskSpecMap["properties"].(map[string]interface{})
+	// iterate over the inputMap and fill in the default values
+	for key, value := range inputMap {
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
+			// fmt.Println("value is not a map", key)
```
Comment (chuang8511) [inline application/github/v0/main.go:125]: `// fmt.Println("value is not a map", key)`
Please take them out.

Comment (chuang8511) [reply application/github/v0/main.go:125]: NIT
About memo, my preference is that we should make the code readable itself rather than using memo to explain code.

The memo is for the special decision or case.
So, I will not add those memo in the codebase.

`// iterate over the inputMap and fill in the default values`

**Review by chuang8511 — COMMENTED** [review]
second draft review.
I will check in details next week.

File: application/github/v0/review_coment.go [hunk 29]
```diff
@@ -0,0 +1,151 @@
+package github
```
Comment (donch1989) [inline application/github/v0/review_coment.go:1]: The filename has typo.

File: application/github/v0/utils.go [hunk 30]
```diff
@@ -0,0 +1,35 @@
+package github
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/instill-ai/x/errmsg"
+)
+
+func middleWare(req string) int {
```
Comment (donch1989) [inline application/github/v0/utils.go:10]: What is the purpose of this function?

File: application/github/v0/issue_service_test.go [hunk 31]
```diff
@@ -0,0 +1,112 @@
...
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v62/github"
+)
+
+type MockIssuesService struct{}
+
+func (m *MockIssuesService) ListByRepo(ctx context.Context, owner, repo string, opt *github.IssueListByRepoOptions) ([]*github.Issue, *github.Response, error) {
+	switch middleWare(owner) {
```
Comment (donch1989) [inline application/github/v0/issue_service_test.go:13]: Why can the owner string be converted to an HTTP code?

**Review by donch1989 — CHANGES_REQUESTED** [review]

File: application/github/v0/config/definition.json (hunk 27 above)
[the remark this answers is not in the document]
Comment (YCK1130) [reply application/github/v0/config/definition.json:9]: Got it!

One more thing, there are page, page_number options to determine how many items would be returned, should I include these in the field?

File: application/github/v0/main.go [hunk 32]
```diff
@@ -0,0 +1,179 @@
...
+		return nil, errmsg.AddMessage(
+			fmt.Errorf("not supported task: %s", task),
+			fmt.Sprintf("%s task is not supported.", task),
+		)
+	}
+
+	return &base.ExecutionWrapper{Execution: e}, nil
+}
+
+func (e *execution) fillInDefaultValues(input *structpb.Struct) (*structpb.Struct, error) {
```
[the remark this answers is not in the document]
Comment (YCK1130) [reply application/github/v0/main.go:101]: That is a good idea. However, there are some concerns described in the Slack.

File: application/github/v0/config/tasks.json [hunk 33]
```diff
@@ -0,0 +1,1270 @@
...
+          "type": "string"
+        },
+        "body": {
+          "description": "Body of the PR",
+          "instillFormat": "string",
+          "instillUIOrder": 5,
+          "title": "PR body",
+          "type": "string"
+        },
+        "diff_url": {
```
Comment (donch1989) [inline application/github/v0/config/tasks.json:41]:
```suggestion
        "diff-url": {
```
Please make sure all property keys are in kebab-case.

File: application/github/v0/config/tasks.json [hunk 34]
```diff
@@ -0,0 +1,1270 @@
...
+          "type": "string"
+        },
+        "base": {
+          "description": "base commit of the PR (in SHA value)",
+          "instillFormat": "string",
+          "instillUIOrder": 9,
+          "title": "PR base",
+          "type": "string"
+        },
+        "comments_num": {
```
Comment (donch1989) [inline application/github/v0/config/tasks.json:62]:
```suggestion
        "comments-num": {
```

File: application/github/v0/config/tasks.json [hunk 35]
```diff
@@ -0,0 +1,1270 @@
...
+          "description": "number of comments on the PR",
+          "instillFormat": "integer",
+          "instillUIOrder": 10,
+          "title": "Number of PR comments",
+          "type": "integer"
+        },
+        "commits_num": {
```
Comment (donch1989) [inline application/github/v0/config/tasks.json:69]:
```suggestion
        "commits-num": {
```

**Review by donch1989 — CHANGES_REQUESTED** [review]
Please update all property keys to kebab-case.

**Comment by donch1989** [comment]
Hi @YCK1130
Could you fix the conflict?
